R/cmd_methods.R

```diff
@@ -127,9 +127,19 @@ monocle_create <- function(
             else if (toupper(substr(file, nchar(file)-2, nchar(file))) == 'MTX')
                 assign(var, Matrix::readMM(file))
             else if (toupper(substr(file, nchar(file)-2, nchar(file))) == 'TSV')
+            {
                 assign(var, as.matrix(read.delim(file, row.names = 1, stringsAsFactors = FALSE)))
+                #dimensionality check to account for the possible lack of a header
+                if (dim(get(var))[1] != dim(expression_matrix)[2])
```

Comment:
Always compare to the `dim(expression_matrix)[2]`? Does it work for both `genes.tsv` and `barcodes.tsv`?

Reply:
Good catch, my bad.  Will fix.

Reply:
Guys, why are you just ignoring Andrey's PR, which we were improving, in favour of this completely independent fix?

Reply:
I believe we've now accounted for the issues in https://github.com/ebi-gene-expression-group/monocle-scripts/pull/7. Of course communication on that PR earlier on might have headed off the inadvertent 'competition' ;-)

Reply:
I'm already a bit overwhelmed working with the fancy environments/wrappers/tests etc. Ni has going on, so I felt more comfortable attempting to fix the issue in my code rather than try to work with the modified approach of the PR. Sorry about that. For what it's worth, it's entirely possible that your PR will be the one that gets ultimately accepted.

Reply:
Don't get me wrong, I'm not precious about the fix- don't care as long as the issues are addressed and the code is maintainable. Communication would have been nice is all.